### PR TITLE
Add documentation on name restrictions.

### PR DESCRIPTION
**Pull request by songy23**
For View and Measure.

### CODE REVIEW

File: core/src/main/java/io/opencensus/stats/View.java [hunk 1]
```diff
@@ -75,7 +77,8 @@
   /**
    * Constructs a new {@link View}.
    *
-   * @param name the {@link Name} of view. Must be unique.
+   * @param name the {@link Name} of view. Must be unique. Suggested format for name:
+   *     {@code <web_host>/<path>}.
```
Comment (sebright) [inline core/src/main/java/io/opencensus/stats/View.java]: I think the suggested format should be specified on Name.create, and Name.create should validate the String.

File: core/src/main/java/io/opencensus/stats/View.java [hunk 2]
```diff
@@ -37,7 +37,10 @@
   }
 
   /**
-   * Name of view. Must be unique.
+   * Name of view. Must be unique. Should be a ASCII string with a length no greater than 255
+   * characters.
+   *
+   * <p>Suggested format for name: {@code <web_host>/<path>}.
```
Comment (sebright) [inline core/src/main/java/io/opencensus/stats/View.java]: See my next comment.  The detail that you added should probably go in the Name class.

File: core/src/main/java/io/opencensus/stats/Measure.java [hunk 3]
```diff
@@ -66,6 +69,11 @@ private Measure() {
...
     /**
      * Constructs a new {@link MeasureDouble}.
+     *
+     * @param name name of {@code Measure}. Suggested format: {@code <web_host>/<path>}.
+     * @param description description of {@code Measure}.
+     * @param unit unit of {@code Measure}.
+     * @return a {@code MeasureDouble}.
      */
     public static MeasureDouble create(String name, String description, String unit) {
       // TODO(dpo): ensure that measure names are unique, and consider if there should be any
```
Comment (sebright) [inline core/src/main/java/io/opencensus/stats/Measure.java]: This method should validate the String, using StringUtil.  You could also remove part of the TODO.

File: core/src/main/java/io/opencensus/stats/Measure.java [hunk 4]
```diff
@@ -97,6 +105,11 @@ public static MeasureDouble create(String name, String description, String unit)
...
     /**
      * Constructs a new {@link MeasureLong}.
+     *
+     * @param name name of {@code Measure}. Suggested format: {@code <web_host>/<path>}.
+     * @param description description of {@code Measure}.
+     * @param unit unit of {@code Measure}.
+     * @return a {@code MeasureLong}.
      */
     public static MeasureLong create(String name, String description, String unit) {
       // TODO(dpo): ensure that measure names are unique, and consider if there should be any
```
Comment (sebright) [inline core/src/main/java/io/opencensus/stats/Measure.java]: Same here

File: core/src/main/java/io/opencensus/stats/Measure.java (hunk 3 above)
Comment (songy23) [inline core/src/main/java/io/opencensus/stats/Measure.java]: Done.

File: core/src/main/java/io/opencensus/stats/Measure.java (hunk 4 above)
Comment (songy23) [inline core/src/main/java/io/opencensus/stats/Measure.java]: Done.

File: core/src/main/java/io/opencensus/stats/View.java (hunk 2 above)
Comment (songy23) [inline core/src/main/java/io/opencensus/stats/View.java]: Done.

File: core/src/main/java/io/opencensus/stats/View.java (hunk 1 above)
Comment (songy23) [inline core/src/main/java/io/opencensus/stats/View.java]: Done. Also added a few tests.

File: core/src/test/java/io/opencensus/stats/MeasureTest.java [hunk 5]
```diff
@@ -19,31 +19,30 @@
...
   @Test
-  public void testNameMaxLength() {
-    char[] name = new char[StringUtil.MAX_LENGTH];
-    char[] truncName = new char[StringUtil.MAX_LENGTH + 10];
-    Arrays.fill(name, 'n');
-    Arrays.fill(truncName, 'n');
-    assertThat(makeSimpleMeasure(new String(name)).getName())
-        .isEqualTo(makeSimpleMeasure(new String(truncName)).getName());
+  public void preventTooLongMeasureName() {
+    String longName = new String(new char[StringUtil.MAX_LENGTH + 1]).replace('\0', 'a');
```
Comment (songy23) [inline core/src/test/java/io/opencensus/stats/MeasureTest.java]: Updated.

File: core/src/test/java/io/opencensus/stats/ViewTest.java [hunk 6]
```diff
@@ -103,6 +110,20 @@ public void preventNullViewName() {
   }
 
   @Test
+  public void preventTooLongViewName() {
+    String longName = new String(new char[StringUtil.MAX_LENGTH + 1]).replace('\0', 'a');
```
Comment (songy23) [inline core/src/test/java/io/opencensus/stats/ViewTest.java]: Done.

File: core/src/main/java/io/opencensus/stats/Measure.java [hunk 7]
```diff
@@ -31,7 +35,10 @@
       Function<? super MeasureLong, T> p1);
 
   /**
-   * Name of measure, as a {@code String}.
+   * Name of measure, as a {@code String}. Should be a ASCII string with a length no greater than
+   * 255 characters.
+   *
+   * <p>Suggested format for name: {@code <web_host>/<path>}.
```
Comment (adriancole) [inline core/src/main/java/io/opencensus/stats/Measure.java]: add rationale for suggested format? For example, this avoids problems with dots (ex in some backends like elasticsearch, dots hint at an object structure and interfere with indexing). Not sure your rationale..

Comment (songy23) [inline core/src/main/java/io/opencensus/stats/Measure.java]: I think this is following some RPC naming conventions but I'm not sure either. @dinooliva can you answer this question?

File: core/src/main/java/io/opencensus/stats/Measure.java [hunk 8]
```diff
@@ -31,7 +35,10 @@
       Function<? super MeasureLong, T> p1);
 
   /**
-   * Name of measure, as a {@code String}.
+   * Name of measure, as a {@code String}. Should be a ASCII string with a length no greater than
+   * 255 characters.
```
Comment (dinooliva) [inline core/src/main/java/io/opencensus/stats/Measure.java]: Apologies - it should be 256 characters (here and elsewhere).

File: core/src/main/java/io/opencensus/stats/Measure.java (hunk 7 above)
Comment (dinooliva) [inline core/src/main/java/io/opencensus/stats/Measure.java]: Theses are some of our cross-language restrictions - we plan to put a document in place soon with more explanation. Might be worth adding a todo to add a link to the document when it becomes available.

There is still some discussion on this particular point, although details should be resolved soon.

File: core/src/main/java/io/opencensus/stats/Measure.java (hunk 8 above)
Comment (songy23) [inline core/src/main/java/io/opencensus/stats/Measure.java]: Updated

File: core/src/test/java/io/opencensus/stats/ViewTest.java [hunk 9]
```diff
@@ -103,6 +113,22 @@ public void preventNullViewName() {
...
+    View.Name.create(longName);
+  }
+
+  @Test
+  public void preventNonPrintableViewName() {
+    thrown.expect(IllegalArgumentException.class);
+    View.Name.create("\2");
+  }
+
+
```
Comment (dinooliva) [inline core/src/test/java/io/opencensus/stats/ViewTest.java]: nit - extra line

**Pull request merged**